### PR TITLE
Job to clean expired advanced field values

### PR DESCRIPTION
**Pull request by qnkhuat**
Pursuant to #23435, this PR adds a step to clean expired advanced field values. 
This step runs first in our existing job to scan field values.

### CODE REVIEW

File: src/metabase/sync/field_values.clj [hunk 1]
```diff
@@ -45,23 +51,58 @@
...
-(def ^:private field-values-steps
-  [(sync-util/create-sync-step "update-field-values" update-field-values-for-database! update-field-values-summary)])
+(defn- delete-expired-advanced-field-values-summary [{:keys [deleted]}]
+  (trs "Deleted {0} expired advanced fieldvalues" deleted))
+
+(s/defn delete-expired-advanced-field-values-for-table!
+  "Delete all expired advanced FieldValues for a table.
+  For more info about advanced FieldValues, check the docs in [[metabase.models.field-values/field-values-types]]"
+  [table :- i/TableInstance]
+  (reduce (fn [acc field]
```
Comment (bshepherdson) [inline src/metabase/sync/field_values.clj:72]: This inner function could become its own function of just `field` that returns `row-count`, then this outer function becomes
```
(->> (table->fields-to-scan table)
     (map the-extracted-function)
     (reduce +))
```
which I think is clearer.

File: src/metabase/sync/field_values.clj [hunk 2]
```diff
@@ -45,23 +51,58 @@
...
+                (+ acc rows-count))))
+          0
+          (table->fields-to-scan table)))
+
+(s/defn ^:private delete-expired-advanced-field-values-for-database!
+  [_database :- i/DatabaseInstance
+   tables :- [i/TableInstance]]
+  {:deleted (apply + (map delete-expired-advanced-field-values-for-table! tables))})
+
+(defn ^:private make-sync-field-values-steps
```
[the remark this answers is not in the document]
Comment (qnkhuat) [reply src/metabase/sync/field_values.clj:92]: yes, I believe we mostly use ^:private for `s/defn`.

I'll change it to `defn-`

File: test/metabase/sync/field_values_test.clj [hunk 3]
```diff
@@ -54,6 +54,59 @@
...
+             :hash_key   "random-hash"
+             :created_at expired-created-at
+             :updated_at expired-created-at}
+            ;; new full fieldvalues
+            {:field_id   field-id
+             :type       "full"
+             :created_at now
+             :updated_at now}])]
+      (is (= (repeat 2 {:deleted 2})
+             (sync-database!' "delete-expired-advanced-field-values" (data/db))))
```
[the remark this answers is not in the document]
Comment (qnkhuat) [reply test/metabase/sync/field_values_test.clj:106]: Do you mean the `old-full-values` entry that has hash-key? if it is then yes it shouldn't have a hash-key. I've removed it.

File: src/metabase/sync/field_values.clj (hunk 1 above)
Comment (qnkhuat) [reply src/metabase/sync/field_values.clj:72]: it is!
Updated per your suggestion.

**Pull request merged**